src/Microsoft.Health.Dicom.SqlServer/Features/Query/SqlQueryGenerator.cs

```diff
@@ -126,8 +128,7 @@ private void AppendFilterTable(string filterAlias)
 
             if ((int)_schemaVersion >= SchemaVersionConstants.SupportDataPartitionSchemaVersion)
             {
-                // TODO: Actual PartitionKey should be passed as a filter condition
-                _stringBuilder.AppendLine($"AND {StudyTableAlias}.{VLatest.Study.PartitionKey} = {DefaultPartition.Key}");
+                _stringBuilder.AppendLine($"AND {StudyTableAlias}.{VLatest.Study.PartitionKey} = {_partitionKey}");
```

Comment:
When we were doing our initial investigation, having here was enough. Ran some more scenarios and added partition to the join condition where it is used as filter column indexes. 

As far as extended query tag, currently we seek on the tag value and tag key predicate and join on series key, study key and instance key. 

We have AB#86014 to revisit the extended query indexes.

Reply:
@smithago @bcarthic:  Here's an overview of QIDO query plan performance between V5 and V6:

| Query scenario | Comparison |
| --- | --- |
| Search for studies | Similar |
| Search for series | Similar |
| Search for series within a study | Similar |
| Search for instances | V6 eliminates a nested loop | 
| Search for XQT | V6 eliminates a nested loop |
| Search for instances within a study | V6 eliminates a nested loop |
| Search for instances within a study and series | V6 eliminates a nested loop |

> `.sqlplan` files for each scenario are included as attachments on [ADO work item 86268](https://microsofthealth.visualstudio.com/Health/_workitems/edit/86268) since they can't be attached to GitHub comments.

The biggest existing issue with all QIDO queries is that indexes need to be updated to include watermark as a key column rather than an included column to avoid index scans. As Karthik mentioned above, we will address this in a follow-up PR.